flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/plan/common/ViewsExpandingTest.scala

```diff
@@ -91,6 +91,39 @@ class ViewsExpandingTest(tableTestUtil: TableTestBase => TableTestUtil) extends
     tableUtil.verifyPlan(query)
   }
 
+  @Test
+  def testViewExpandingWithMismatchRowType(): Unit = {
+    val tableUtil = tableTestUtil(this)
+    val tableEnv = tableUtil.tableEnv
+    tableUtil.addDataStream[(Int, String, Int)]("t1", 'a, 'b, 'c)
+    def createAggSqlView(originTable: String): CatalogView = {
+      new CatalogViewImpl(
+        s"select a, b, count(c) from $originTable group by a, b",
+        s"select a, b, count(c) from $originTable group by a, b",
+        TableSchema.builder()
+          .field("a", DataTypes.INT().notNull()) // Change the nullability intentionally.
+          .field("b", DataTypes.STRING())
+          .field("c", DataTypes.INT())
+          .build(),
+        new util.HashMap[String, String](),
+        ""
+      )
+    }
+    val catalog = tableEnv.getCatalog(tableEnv.getCurrentCatalog).get()
```

Comment:
Let's test one thing at a time. We don't need to go through multiple views to test types casting.

Reply:
Sure, let me simplify the test.